resolwe_bio/tools/expression_fpkm_tpm.R

```diff
@@ -12,31 +13,36 @@ args = parser$parse_args(commandArgs(trailingOnly=TRUE))
 
 # retrieve exon lengths
 annot <- import.gff(args$annotation, format=args$format, genome="NA", feature.type=args$feature_type)
-exon_lengths=width(annot)
-names(exon_lengths)=elementMetadata(annot)[,args$id_attribute]
+exon_lengths <- width(annot)
+names(exon_lengths) <- elementMetadata(annot)[, args$id_attribute]
 
-# calculate gene lengths
-exon_lengths_by_gene=split(exon_lengths, names(exon_lengths))
-gene_lengths=sapply(exon_lengths_by_gene, sum)
+# calculate gene lengths, note that the result is lexicographically sorted
```

Comment:
Change to: # calculate gene or transcript lengths

Reply:
Done.